File: onboarding/src/indexing-relevance/scala/src/main/scala/com/algolia/Relevance.scala
package com.algolia

import algolia.AlgoliaClient
import algolia.AlgoliaDsl._
import algolia.objects.{IndexSettings, Ranking, SearchableAttributes}

object Relevance {

  def main(args: Array[String]): Unit = {
    val client = new AlgoliaClient("YourApplicationID", "YourAPIKey")

    val settings = IndexSettings(
      searchableAttributes = Some(Seq(
        SearchableAttributes.attribute("searchable(brand)"),
        SearchableAttributes.attribute("type"),
        SearchableAttributes.attribute("name"),
        SearchableAttributes.attribute("categories"),
        SearchableAttributes.attribute("description")
      )),
      customRanking = Some(Seq(
        Ranking.attribute("popularity")
      )),
      attributesForFaceting = Some(Seq(
        "categories",
        "brand",
        "price"
      ))
    )

    for {
      result <- client.execute(changeSettings of "myIndex" `with` settings)
    } yield result
  }

}

File: onboarding/src/indexing-relevance/java/src/main/java/com/algolia/Relevance.java
package com.algolia;

import com.algolia.search.APIClient;
import com.algolia.search.ApacheAPIClientBuilder;
import com.algolia.search.Index;
import com.algolia.search.exceptions.AlgoliaException;
import com.algolia.search.objects.IndexSettings;

import java.util.Arrays;
import java.util.Collections;

public class Relevance {

  public static void main(String[] args) throws AlgoliaException {
    APIClient client =
      new ApacheAPIClientBuilder("YourApplicationID", "YourAdminAPIKey").build();

    Index<Product> index = client.initIndex("demo_ecommerce", Product.class);
    index.setSettings(
      new IndexSettings()
        .setSearchableAttributes(Arrays.asList(
          "searchable(brand)",
          "type",
          "name",
          "categories",
          "description"
        ))
        .setCustomRanking(Collections.singletonList("popularity"))
        .setAttributesForFaceting(Arrays.asList(
          "categories",
          "brand",
          "price"
        ))
    );
  }
}
